src/glossary.md

```diff
@@ -0,0 +1,281 @@
+
+## Bytecode
+
+Bytecode is machine-readable code, usually generated from a compiler.
+
+## Chain
+
+Another name for a blockchain.
+
+## ChainId
+
+A unique id for a blockchain.
+
+## Client
+
+The Fuel client refers to the software that runs the Fuel Virtual Machine. It can be used to validate and produce blocks, and send transactions.
+
+## Coinbase
+
+Coinbase refers to the validators paying themselves for processing a block from the transaction fees. Having a coinbase transaction on each block makes this process transparent to all users.
+
+## Consensus
+
+The consensus layer defines the state and validates that all nodes on the blockchain have the same state.
+
+## Consensus Parameters
+
+Consensus parameters are the rules used by clients to determine and validity of and finalize a block.
+
+## Contract Call
+
+Calling a contract means invoking a function from a smart contract that has been deployed to the blockchain.
+
+## Data Availability
+
+The data availability layer ensures that block data has been published to the network.
+
+## EIP
+
+EIP stands for Ethereum Improvement Proposal. It refers to a proposal to upgrade the core software powering the Ethereum blockchain.
+
+## EOA
+
+EOA stands for Externally Owned Account. It refers to a wallet address that is not controlled by a contract.
+
+## EVM
+
+EVM stands for Ethereum Virtual Machine, which is the virtual machine used for the Ethereum network.
+
+## Execution
+
+Execution refers to the processing of transactions by nodes in a network.
+
+## Faucet
+
+A faucet is a service that provides free tokens for a testnet.
+
+## Forc
+
+Forc is short for Fuel Orchestrator. Similar to Cargo for Rust, Forc is the build system and package manager for Sway. It is used to build, test, and deploy Sway contracts.
+
+## Fraud Proof
+
+Fraud proofs are a blockchain verification mechanism whereby a claim on a new block is accepted unless a proof the claim is invalid is provided within some configurable time window. Both the Fuel protocol and the FuelVM are designed to be fraud-provable in restrictive environments such as the Ethereum Virtual Machine.
+
+## Fuel
+
+The Fuel blockchain.
+
+## Fuels
+
+Fuels is the name of the Fuel Rust and Typescript SDKs used to interact with a contract, similar to ethers.js or web3.js
+
+## Fuelup
+
+Fuelup is the official toolchain and package manager for the Fuel toolchain.
+
+## FuelVM
+
+The FuelVM is the virtual machine powering the Fuel blockchain.
+
+## Fuel Core
+
+`fuel-core` is the name of the Fuel client implementation.
+
+## Gas
+
+Gas is a variable fee charged by a node to process a transaction that is executed on-chain.
+
+## Indexer
+
+An indexer is a program that watches and organizes blockchain data so it can be easily queried.
+
+## Input
+
+An input refers to a transaction input, or which state elements (usually: UTXOs) are consumed by a transaction.
```

Comment:
```suggestion
An input refers to a transaction input, which is a UTXO consumed by a transaction.
```
Fuel has Contract UTXOs, rather than state-element UTXOs. When contract state is read or written, the whole Contract UTXO is consumed.